src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -37,6 +37,14 @@ public void EnsureInitialized(Workspace workspace, string projectRazorJsonFileNa
 
     public void NotifyDynamicFile(ProjectId projectId)
     {
+        // Since there is no "un-notify" API to indicate that callers no longer care about a project, it's entirely
+        // possible that by the time we get notified, a project might have been removed from the workspace. Whilst
+        // that wouldn't cause any issues we may as well avoid creating a task scheduler.
+        if (_workspace is null || !_workspace.CurrentSolution.ContainsProject(projectId))
```

Comment:
Is there any concern we might get this call before the _workspace has the indicated project?

Reply:
Fortunately that is not possible